src/app/home-page/home-page.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, OnInit } from '@angular/core';
+import { Subject } from 'rxjs';
+
+import { CampaignService, SearchQuery } from '../campaign.service';
+import { CampaignSummary } from '../campaign-summary.model';
+
+@Component({
+  selector: 'app-home-page',
+  templateUrl: './home-page.component.html',
+  styleUrls: ['./home-page.component.scss'],
+})
+export class HomePageComponent implements OnInit {
+  public campaigns: CampaignSummary[];
+  public loading = false; // Server render gets initial result set; set true when filters change.
+  public resetSubject: Subject<void> = new Subject<void>();
+
+  private perPage = 6;
+  private query: {[key: string]: any};
+
+  constructor(
+    private campaignService: CampaignService,
+  ) {
+  }
+
+  ngOnInit() {
+    this.setDefaults();
+    this.run();
+  }
+
+  setDefaults() {
+    this.query = {
+      limit: this.perPage,
+      offset: 0,
+      sortDirection: 'desc',
+      sortField: 'matchFundsRemaining',
+    };
+  }
+
+  private run() {
+    this.campaigns = [];
+    this.loading = true;
+
+    this.campaignService.search(this.query as SearchQuery).subscribe(campaignSummaries => {
+      this.campaigns = campaignSummaries; // Success
+      this.loading = false;
+    }, () => {
+        this.loading = false;
+      },
+    );
+  }
+
+  search(term: string) {
+    // TODO DON-303 redirect to /explore page with query param
```

Comment:
I don't think we can keep the button in the template without this working properly, so we need to take it out for now for this to be safe to merge.